test/unit/services/csv_preview_service_test.rb

```diff
@@ -1,6 +1,8 @@
 require "test_helper"
 
-class CsvPreviewControllerTest < ActionController::TestCase
+class CsvPreviewServiceTest < ActionController::TestCase
```

Comment:
This is a bit of a confused spec...see docs here for what the [ActionController::TestCase]( https://api.rubyonrails.org/classes/ActionController/TestCase.html) is meant to test. It also seems to be testing all the private methods, which again is not conventional (or particularly informative)

I think in this PR we should at a minimum implement a unit test for the `CSVPreviewService` that only tests the public `csv_rows` method. We can involve the same csv data types to make sure we aren't losing coverage.

Stretch goal:

- a `CsvPreviewServiceControllerTest < ActionController::TestCase ` test, that is actually a controller test. Currently we don't have a controller test at all, just a massive [integration test](https://github.com/alphagov/frontend/blob/main/test/integration/csv_preview_test.rb). Probably the easiest thing to do is to [cherry pick from that integration test](https://github.com/alphagov/frontend/blob/main/test/integration/csv_preview_test.rb) . But this is a nice to have, as we do have test coverage in the integration test. So we could do this later/in a separate piece of work if you don't want to do it now.

Reply:
Argh, I missed that.

Reply:
OK, I'll add a more complete unit test for the CSVPreviewService.

Reply:
Added.